azurerm/internal/services/machinelearning/machine_learning_compute_cluster_resource.go

```diff
@@ -0,0 +1,438 @@
+			name, workspaceID.Name, workspaceID.ResourceGroup)
+	}
+
+	id, err := parse.ComputeClusterID(*resp.ID)
+	if err != nil {
+		return err
+	}
+	d.SetId(id.ID())
+
+	return resourceComputeClusterRead(d, meta)
+}
+
+func resourceComputeClusterRead(d *schema.ResourceData, meta interface{}) error {
+	mlComputeClient := meta.(*clients.Client).MachineLearning.MachineLearningComputeClient
+	mlWorkspacesClient := meta.(*clients.Client).MachineLearning.WorkspacesClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.ComputeClusterID(d.Id())
+	if err != nil {
+		return fmt.Errorf("error parsing Compute Cluster ID `%q`: %+v", d.Id(), err)
+	}
+
+	computeResource, err := mlComputeClient.Get(ctx, id.ResourceGroup, id.WorkspaceName, id.ComputeName)
+	if err != nil {
+		if utils.ResponseWasNotFound(computeResource.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("error making Read request on Compute Cluster %q in Workspace %q (Resource Group %q): %+v",
+			id.ComputeName, id.WorkspaceName, id.ResourceGroup, err)
+	}
+
+	d.Set("name", id.ComputeName)
+
+	workspaceResp, err := mlWorkspacesClient.Get(ctx, id.ResourceGroup, id.WorkspaceName)
+	if err != nil {
+		return err
+	}
+	d.Set("machine_learning_workspace_id", workspaceResp.ID)
+
+	// use ComputeResource to get to AKS Cluster ID and other properties
+	computeCluster, isComputeCluster := (machinelearningservices.BasicCompute).AsAmlCompute(computeResource.Properties)
+	if !isComputeCluster {
+		return fmt.Errorf("compute resource %s is not an Aml Compute cluster", id.ComputeName)
+	}
+
+	d.Set("vm_size", computeCluster.Properties.VMSize)
+	d.Set("vm_priority", computeCluster.Properties.VMPriority)
+
+	d.Set("scale_settings", flattenScaleSettings(computeCluster.Properties.ScaleSettings))
+
+	d.Set("subnet_resource_id", computeCluster.Properties.Subnet.ID)
+
+	if location := computeResource.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	if err := d.Set("identity", flattenComputeClusterIdentity(computeResource.Identity)); err != nil {
+		return fmt.Errorf("error flattening identity on Workspace %q (Resource Group %q): %+v",
+			id.ComputeName, id.ResourceGroup, err)
+	}
+
+	return tags.FlattenAndSet(d, computeResource.Tags)
+}
+
+func resourceComputeClusterUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).MachineLearning.MachineLearningComputeClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.ComputeClusterID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	update := machinelearningservices.ClusterUpdateParameters{
+		ClusterUpdateProperties: &machinelearningservices.ClusterUpdateProperties{},
+	}
+
+	if d.HasChange("scale_settings") {
+		scaleSettings := d.Get("scale_settings").([]interface{})
+		update.ScaleSettings = expandScaleSettings(scaleSettings)
```

Comment:
```suggestion
		update.ClusterUpdateProperties.ScaleSettings = expandScaleSettings(scaleSettings)
```
This probably wants to become a level lower, which should fix the update problem. Testing locally with your branch now.